src/kodeagent/tools.py

```
@@ -377,35 +384,51 @@ def download_file(url: str, save_filename: str = None) -> str:
-        # Create temp file with proper extension
-        file_ext = Path(save_filename).suffix
-        with tempfile.NamedTemporaryFile(
+        final_path = None
+        f = None
+        if save_dir:
+            try:
+                p = Path(save_dir)
+                p.mkdir(parents=True, exist_ok=True)
+                final_path = p / save_name
+                f = open(final_path, 'wb')
+            except Exception: # pylint: disable=broad-exception-caught
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)